src/backend/joanie/core/flows/order.py

```diff
@@ -284,6 +285,7 @@ def _can_be_state_refunded(self):
         return (
             self.instance.state == enums.ORDER_STATE_REFUNDING
             and not has_installment_paid(self.instance)
+            and has_only_refunded_or_canceled_installments(self.instance)
```

Comment:
IMO those two conditions are overlapping. We could remove the `and not has_installment_paid(self.instance)`

Reply:
In this scenario suggested by @jbpenrath, all remaining 'pending' state payments in the payment schedule should be switched to `canceled` or `refunded` before. I think it's already handle in `_do_on_refund` in the base payment backend.